src/modules/staking/sora/components/ClaimRewardsDialog.vue

```diff
@@ -0,0 +1,210 @@
+          :title="inputTitle"
+          :token="rewardAsset"
+          :value="rewardedFundsFormatted"
+          :disabled="true"
+        />
+      </s-form>
+
+      <s-input v-model="payeeAddress" placeholder="Rewards destination" suffix="s-icon-basic-user-24"></s-input>
+
+      <div class="info">
+        <info-line
+          v-if="stakingInitialized"
+          :label="t('soraStaking.info.allRewards')"
+          :value="rewardedFundsFormatted"
+          :asset-symbol="rewardAsset?.symbol"
+          :fiat-value="rewardedFundsFiat"
+        />
+        <info-line
+          :label="t('networkFeeText')"
+          :label-tooltip="t('networkFeeTooltipText')"
+          :value="networkFeeFormatted"
+          :asset-symbol="xor?.symbol"
+          :fiat-value="getFiatAmountByCodecString(networkFee)"
+          is-formatted
+        />
+      </div>
+
+      <s-button
+        type="primary"
+        class="s-typography-button--large action-button"
+        :loading="parentLoading"
+        :disabled="isInsufficientXorForFee || valueFundsEmpty || isInsufficientBalance"
+        @click="handleConfirm"
+      >
+        <template v-if="isInsufficientXorForFee || isInsufficientBalance">
+          <template v-if="isInsufficientBalance">
+            {{ t('insufficientBalanceText', { tokenSymbol: rewardAsset?.symbol }) }}
+          </template>
+          <template v-if="isInsufficientXorForFee">
+            {{ t('insufficientBalanceText', { tokenSymbol: xor?.symbol }) }}
+          </template>
+        </template>
+        <template v-else-if="valueFundsEmpty">
+          {{ t('buttons.enterAmount') }}
+        </template>
+        <template v-else>
+          {{ t('confirmText') }}
+        </template>
+      </s-button>
+      <div class="check-pending-rewards" @click="checkPendingRewards">
+        Check rewards per era and validator ({{ pendingRewards?.length ?? 0 }})
+      </div>
+    </div>
+  </dialog-base>
+</template>
+
+<script lang="ts">
+import { FPNumber } from '@sora-substrate/util';
+import { components, mixins } from '@soramitsu/soraneo-wallet-web';
+import { Component, Mixins, Watch, Prop } from 'vue-property-decorator';
+
+import { Components } from '@/consts';
+import { lazyComponent } from '@/router';
+
+import StakingMixin from '../mixins/StakingMixin';
+
+import type { CodecString } from '@sora-substrate/util';
+import type { NominatorReward } from '@sora-substrate/util/build/staking/types';
+
+@Component({
+  components: {
+    TokenInput: lazyComponent(Components.TokenInput),
+    DialogBase: components.DialogBase,
+    InfoLine: components.InfoLine,
+  },
+})
+export default class ClaimRewardsDialog extends Mixins(StakingMixin, mixins.DialogMixin, mixins.LoadingMixin) {
+  @Prop({ default: () => true, type: Boolean }) readonly isAdding!: boolean;
+
+  payeeAddress = '';
+  payoutNetworkFee: string | null = null;
+
+  @Watch('payee', { immediate: true })
+  async handlePayeeChange() {
```

Comment:
should it be async?

Reply:
What do you mean?

Reply:
didn't see any awaiting logic inside 'handlePayeeChange' function, may be synchronous I guess